packages/core/src/factory.ts

```diff
@@ -41,34 +43,36 @@ export function factory<Resource, Input, Output, Config extends IPrismConfig>(
     resource: Resource,
     input: Input,
     config: Config
-  ): TE.TaskEither<Error, ResourceAndValidation> =>
-    pipe(
-      sequenceValidation(
-        config.validateRequest ? components.validateInput({ resource, element: input }) : E.right(input),
-        config.checkSecurity ? components.validateSecurity({ resource, element: input }) : E.right(input)
-      ),
-      E.fold(
-        validations => validations as IPrismDiagnostic[],
-        () => []
-      ),
-      validations => TE.right({ resource, validations })
+  ): E.Either<Error, ResourceAndValidation> => {
+    const validations = compact([
+      config.checkSecurity ? components.validateSecurity({ resource, element: input }) : undefined,
+      config.validateRequest ? components.validateInput({ resource, element: input }) : undefined,
+    ]);
+
+    return pipe(
+      eitherSequence(validations),
+      E.fold<NonEmptyArray<IPrismDiagnostic>, unknown, IPrismDiagnostic[]>(identity, () => []),
+      validations => E.right({ resource, validations })
```

Comment:
Why do we even bother in returning `Either` if it cannot return Left? (I am aware that it is not related to this PR)

Reply:
Great question, this can effectively be an `Option` rather than an Either — the reason why it's in this weird situation it's because it will eventually be an Either.

Now that both Proxy and Mock halt the request if it's invalid, with some more refactor this will behave as a regular Either and return 422 when the request is invalid.

I'll write up more details in #1135

Reply:
Sounds good 👍